File: 1797-goal-parser-interpretation/goal-parser-interpretation.py
class Solution(object):
    def interpret(self, command):
        """
        :type command: str
        :rtype: str
        """
        i = 0
        new = ""
        while i < len(command):
            if command[i]==")":
                pass
            elif command[i]!="(":
                new+=command[i]
            elif command[i]+command[i+1]=="()":
                new+="o"
                i += 1
            i += 1
        return new

File: 1797-goal-parser-interpretation/goal-parser-interpretation.cs
public class Solution {
    public string Interpret(string command) {
        int i = 0;
        string word = "";
        while(i < command.Length)
        {
            if(command[i]=='G')
            {
                word += "G";
                i++;
            }
            else if(i+1 < command.Length && command.Substring(i, 2)=="()")
            {
                word += "o";
                i+=2;
            }
            else if(i+3 < command.Length && command.Substring(i, 4)=="(al)")
            {
                word += "al";
                i += 4;
            }
            else
            {
                i++;
            }
        }
        return word;
    }
}